src/formatDate.js

```diff
@@ -8,7 +8,36 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let resultDate = '';
+  const objectDateFormat = {};
+  const dateSplitted = date.split(fromFormat[fromFormat.length - 1]);
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    const less = dateSplitted[i] < 30;
+    const more = dateSplitted[i] >= 30;
+
+    if (fromFormat[i] === 'YY' && toFormat.includes('YYYY') && less) {
+      objectDateFormat.YYYY = '20' + dateSplitted[i];
+    } else if (fromFormat[i] === 'YY' && toFormat.includes('YYYY') && more) {
+      objectDateFormat.YYYY = '19' + dateSplitted[i];
+    }
+
+    if (fromFormat[i] === 'YYYY' && toFormat.includes('YY')) {
+      objectDateFormat.YY = dateSplitted[i][2] + dateSplitted[i][3];
+    } else {
+      objectDateFormat[fromFormat[i]] = dateSplitted[i];
```

Comment:
The 'else' block will incorrectly overwrite the 'YYYY' key in 'objectDateFormat' if both 'YYYY' and 'YY' are present in 'fromFormat'. This block should only execute if the current 'fromFormat[i]' is neither 'YY' nor 'YYYY'.